extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -406,8 +407,9 @@ export class JupyterServerInstallation {
 	private async installOfflinePipDependencies(): Promise<void> {
 		let installJupyterCommand: string;
 		if (process.platform === constants.winPlatform) {
+			let cmdOptions = this._usingExistingPython ? '--user' : '--ignore-installed';
```

Comment:
If you're using this same logic multiple places, could / should you be putting in a common function?

Reply:
The refactor might take a while, since there are so many different options floating around. I'd like to use that extra time for testing, if that's all right with you.